### ISSUE TITLE
First character inputted into password field is duplicated

### ISSUE THREAD

**Issue by phume03** [mozilla-mobile/fenix#22992]
### Steps to reproduce

1. Open Firefox on Android
2. Navigate to any page that requires a login (i.e. password and username) and has the reveal password option next to or in the password form field
3. Enter a username and note your entry
4. Enter a password and note your entry
5. Select reveal password and note the result of your previous entry

### Expected behaviour

The password should be as entered e.g. Password2021+December

### Actual behaviour

The first character in the form field is duplicated without any warning so your result is actually:

PPassword2021+December

This only happens in password fields in web forms. So if you do not note it, you may keep entering your password and be informed that your password is wrong when in actual fact your believe that you are typing the right password. But the bug in the application or android system is duplicating some characters. Resulting in an incorrect password being submitted. This behaviour might be carried onto registration form password or hidden fields. Thus, if you do not check, the password you enter may be different than what you submit!

### Android version

Android 7.1.1

### Firefox release type

Firefox

### Firefox version

93.1.0

### Additional information

This error was first noted on Google Chrome on Android. Switching the app initially helped but the bug caught up eventually! It was unreported to the Chrome dev team.

Chrome had other bugs in addition to this one and they will be reported to the Chrome team. But incase they have caught up, these could include duplicating the first three characters typed in messaging platforms like Facebook, online and not the app; duplication of entire words; dictionary correction errors (dictionary suggestions not picked up even when chosen); and so on. While these were suspected to be Chrome bugs, because the errors involved the dictionary it was also suspected that they could be Android system bugs! Hence the need to change the browser and intentionally block known updates!

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FNXV2-19003)

**Comment by Mugurell** [comment]
@pumie03 Thank you for the report!
Tried reproducing on https://jsfiddle.net/59fgjtz4/show but I was unable to:

https://user-images.githubusercontent.com/11428869/147471805-7e701dd4-1572-4609-b86a-579241d7227b.mp4

Maybe it's an issue for just a specific websites?
Also seeing that you're using a version a bit older of Firefox. Can you please update to the latest from Playstore and if the issue still reproduces share the exact website on which you've seen this issue?
A screen recording would also help greatly.

**Comment by sheikh-azharuddin** [comment]
no issue found with latest firefox..I believe it might be keyboard issue ..example in the past Microsoft SwiftKey was causing issue with some fields ..Google gboard had no issue

**Comment by SoftVision-LorandJanos** [comment]
Hi @pumie03! I couldn't reproduce your issue either.
Please try this on a current version, if you can, and maybe upload a video of your issue, as mentioned above. Thank you!
Tested on the latest Nightly (2021-12-31) and latest Release (95.2.0).
Device used: Xiaomi Mi11 Lite (Android 11).

**Comment by phume03** [comment]
Hi @SoftVision-LorandJanos , @sheikh-azharuddin , and @Mugurell ,

I took the screenshot below on Facebook, on Firefox  99.2.0 (Build #2015874211), 3fbd94a59+
AC: 99.0.10, 2cf4dbe50f
GV: 99.0.1-20220411174855
AS: 91.1.1

The issue has persisted through updates and I have even made it a part of some of my passwords just because it persists and it is annoying to type and remove a character each time. It takes a great deal of patience! Besides, the extra character, although duplicated, just adds to my password strength if accounted for when the password is set!

Thank you for your responses thus far. And the issue is not isolated to Facebook as in the screenshot I have provided. I cannot provide a video — in more detail: the duplicated character only appears after the second character is typed, and if the field is hidden and you are not paying attention, you might miss the extra dot. It does not occur in the email field. Also, specifically for Facebook, this issue does not persist on the app. This possibly rules out a keyboard issue (or Facebook issue as this occurs on other sites, I believe github is one where such does not occur) and points to a browser issue/hack/bug.

![Screenshot_20220722-091351](https://user-images.githubusercontent.com/96667636/180384693-b1e4bba3-09c0-4f65-88ca-969349422c04.png)
